### ISSUE TITLE
Docker client needs authentication to access protected or private images

### ISSUE THREAD

**Issue by MichaelRoeder** [hobbit-project/platform#2]
The Platform can not pull benchmark or system images that are protected or private in Gitlab.
```
2017-03-14 10:13:15,330 INFO [com.spotify.docker.client.LoggingPullHandler] - <pull git.project-hobbit.eu:4567/XXXXX:latest: ProgressMessage{id=hobbit-worker5, status=Pulling git.project-hobbit.eu:4567/XXXXX:latest..., stream=null, error=null, progress=null, progressDetail=ProgressDetail{current=0, start=0, total=0}}>
...
2017-03-14 10:13:15,525 INFO [com.spotify.docker.client.LoggingPullHandler] - <pull git.project-hobbit.eu:4567/XXXXX:latest: ProgressMessage{id=hobbit-worker5, status=Pulling git.project-hobbit.eu:4567/XXXXX:latest... : Error response from daemon: Get https://git.project-hobbit.eu:4567/v2/XXXXX/manifests/latest: denied: access forbidden, stream=null, error=null, progress=null, progressDetail=ProgressDetail{current=0, start=0, total=0}}>
```
The Docker client used in the platform needs credentials to pull non-public images. It is not sufficient to give these credentials to the Docker daemon that is used by the client.

Solution: provide credentials via environmental variables that are used by the Docker client.

**Comment by MichaelRoeder** [comment]
The Problem seems to be bound to the docker swarm since it can not be reproduced using a local Docker installation.

**Comment by MichaelRoeder** [comment]
fixed in v1.0.1
